src/TLuaInterpreter.cpp

```diff
@@ -1314,27 +1270,15 @@ int TLuaInterpreter::selectCaptureGroup(lua_State* L)
 int TLuaInterpreter::getLines(lua_State* L)
 {
     int n = lua_gettop(L);
-    int s = 0;
+    int s = 1;
     QString windowName;
     if (n > 2) {
-        if (!lua_isstring(L, ++s)) {
-            lua_pushfstring(L, "getLines: bad argument #%d type (mini console, user window or buffer name as string expected {may be omitted for the \"main\" console}, got %s!)", s, luaL_typename(L, s));
-            return lua_error(L);
-        }
-        windowName = lua_tostring(L, s);
-    }
-
-    if (!lua_isnumber(L, ++s)) {
-        lua_pushfstring(L, "getLines: bad argument #%d type (start line as number expected, got %s!)", s, luaL_typename(L, s));
-        return lua_error(L);
-    }
-    int lineFrom = lua_tointeger(L, s);
-
-    if (!lua_isnumber(L, ++s)) {
-        lua_pushfstring(L, "getLines: bad argument #%d type (end line as number expected, got %s!)", s, luaL_typename(L, s));
-        return lua_error(L);
+        windowName = getVerifiedString(L, __func__, 1, "mini console, user window or buffer name {may be omitted for the \"main\" console}", true);
```

Comment:
Doesn't the `true` argument cause some text about this being optional to be included - which is not what the original said exactly - and which won't read quite right with this proposed text?

Reply:
Original:
`getLines: bad argument #%d type (mini console, user window or buffer name as string expected {may be omitted for the \"main\" console}, got %s!`

New:
`getLines: bad argument #%d type (mini console, user window or buffer name {may be omitted for the \"main\" console} as string is optional, got %s!``

Yeah I am not sure about all the {comments} you all sprinkled into these sentences in position X or Y for now I just append them to the name and see where we land:

`%s: bad argument #%d type (name as {X} string expected, got %s!`
`%s: bad argument #%d type (name as string expected {Y}, got %s!`

I say drop the comment in brackets, but may be differing opinions.

Reply:
Because for the {X} case, some of the functions have different targets - one or two of them also work for *labels* and some aren't relevant for *buffers*, that sort of thing...

Reply:
Actually I find these "may be omitted" comments should stay in the wiki and not the error messages. They are far and few in between, most error messages don't have them at all, and I think it's for a reason.
If you really want to put them in the client somewhere, how about we create a help command, so help(getlines) will print the manual to the player alright?